macros/stitch_user_identifiers.sql

```diff
@@ -4,21 +4,50 @@
 
 {% macro default__stitch_user_identifiers(enabled, relation=this, user_mapping_relation='snowplow_web_user_mapping') %}
     {% if enabled | as_bool() %}
-      -- Update sessions table with mapping
+
+      {% if var('snowplow__page_view_stitching') %}
+        alter table if exists {{ this }} add column if not exists stitched_user_id {{ type_string() }};
+      {% endif %}
+
+      -- Update sessions /page_views table with mapping
       update {{ relation }} as s
       set stitched_user_id = um.user_id
       from {{ ref(user_mapping_relation) }} as um
-      where s.domain_userid = um.domain_userid;
+      where s.domain_userid = um.domain_userid
+
+      {% if not var('snowplow__stitch_lookback_days') == 0 %}
+
+        {%- set lower_limit, upper_limit = snowplow_utils.return_limits_from_model( this, 'start_tstamp',
+                                                                          'end_tstamp') %}
+        {% set stitch_lookback_limit = get_stitch_lookback_limit() %}
+        and start_tstamp > {{ stitch_lookback_limit }}
+      {% endif %}
+
+
     {% endif %}
 {%- endmacro -%}
 
 {% macro spark__stitch_user_identifiers(enabled, relation=this, user_mapping_relation='snowplow_web_user_mapping') %}
     {% if enabled | as_bool() %}
-      -- Update sessions table with mapping
+
+      {% if var('snowplow__page_view_stitching') %}
+        alter table if exists {{ this }} add column if not exists stitched_user_id {{ type_string() }};
+      {% endif %}
+
+      -- Update sessions /page_views table with mapping
       merge into {{ relation }} as s
       using {{ ref(user_mapping_relation) }} as um
       on s.domain_userid = um.domain_userid
+
+      {% if not var('snowplow__stitch_lookback_days') == 0 %}
+        {%- set lower_limit, upper_limit = snowplow_utils.return_limits_from_model( this , 'start_tstamp',
+                                                                          'end_tstamp') %}
```

Comment:
Does this work given it's called in a post-hook and not in the model itself?

Reply:
Also I think this will just return the full limit from the derived table, rather than the incremental part of it? Maybe?

Reply:
I only need the upper limit tbh. This got compiled when I ran it with 12 lookback days (max is 2021-03-02 23:57:04.705 in the table)
<img width="538" alt="image" src="https://github.com/snowplow/dbt-snowplow-web/assets/95634439/7e4faec4-6e57-4f58-b13b-efc46ee89165">

Reply:
Why only the upper? Wouldn't you want to have the lower limit? Also thinking about it I'm not sure this limit makes a _tonne_ of sense to do, the whole point of stitching is to catch older sessions/page views for a user that has only more recently logged in - so by limiting that scan aren't we removing the main point of it?

Reply:
For some use cases it might be useful. E.g: logging tracking only available after a certain time, or when they want to limit scanning for frequent runs. I thought it is easier to conceptualise this for the user to just give a number based on the latest data you are processing rather than having to keep in mind the lookback days as well but we could change the logic.

Reply:
I think if we keep it it should be lower limit (because you want to cover all the new data), and then the lookback applied on top of that to cover previous sessions/pvs that now have userids. Maybe a higher default lookback as well like 365 days?

My only other worry is you then risk getting the same domain_userid with multiple stitched user ids because it was outside the lookback window, but I suppose if we make that clear to people then it's fine.

Reply:
Valid point on the possible duplication issue, by default I wanted to keep it a 0, aka not have this out of the box at all. I'm thinking of removing this altogether for now until we get a request for it.

Reply:
My preference would be we should - users can already disable it on runs and then only enable it for a specific run as needed that should help keep cost down?

Reply:
Enabling / disabling is based on the variable itself not the limit. I assume if they want to rarely run it they would want to run it fully.

Reply:
If we find a viable use case for the limit not just a possible one I hinted at then I can tweak the limit logic so that it is model safe (like you suggested using the lower limit) not just a rough user controlled field like in this first iteration is.